### PR TITLE
Upgrade syndesis version and make API changes to Step

### PR DESCRIPTION
**Pull request by cunningt**
I'm not sure what we want to do here - the following commit changed the Step API in a way that requires changes from the syndesis-extensions actions : 

https://github.com/syndesisio/syndesis/commit/91e424a3c18d105efd6915612b644c6b51c0954e#diff-1ce34ed4cb8daaadfd2ab43eef0d8219L27

I'd like a branch of syndesis-extensions that can be compiled against syndesis 1.4 - can we create a 1.4.x branch of syndesis-extensions and incorporate these changes within?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,7 +28,7 @@
   <description>A Collection of Extensions for Syndesis</description>
 
   <properties>
-    <syndesis.version>1.3.10</syndesis.version>
+    <syndesis.version>1.4.0.fuse-000002</syndesis.version>
```
Comment (zregvart) [inline pom.xml]: I'm not sure about this version change, is this something we wish to do? The previous (I think) kept us in check that we remain compatible from the moment we introduced extensions.

**Comment by zregvart** [comment]
@syndesisio/backend any thoughts here?

**Comment by lburgazzoli** [comment]
The idea was to use the first 7.x tag for compatibility reasons, people that need specific version need to set in in theirs build.

I think the best would be to remove the generic diamond in extension APIs for fuse 7.x and introduce it on 8.x series.

**Comment by squakez** [comment]
This was open since a very long time now. We have already included that change in another work done. Thanks for the PR anyway :)

**Pull request closed without merging**